### ISSUE TITLE
Lint on `const RUNTIME: tokio<Runtime>`

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust-clippy#6088]
### What it does

*What does this lint do?*

It errors on `const` tokio runtimes.

### Categories (optional)

- Kind: Correctness
- Deny by default

*What is the advantage of the recommended code over the original code*

It avoids errors like the following:

```
Io(Custom { kind: Other, error: "reactor gone" })
```

`const` creates a new copy of the runtime every time it's used, while `static` uses the same runtime each time.
I was using this as `RUNTIME.block_on()`, which takes `&mut`, so possibly this could be expanded to any `&mut` use of a `const`.

### Drawbacks

None.

### Example

```rust
pub const RUNTIME: Lazy<Runtime> = Lazy::new(|| Runtime::new().unwrap());
```

should instead be

```rust
pub static RUNTIME: Lazy<Runtime> = Lazy::new(|| Runtime::new().unwrap());
```

**Comment by ebroto** [comment]
@jyn514 would [const_item_mutation](https://github.com/rust-lang/rust/pull/75573) solve your use case? To be landed in 1.48.

**Comment by jyn514** [comment]
Yes it would! Thank you :D https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=5329dc009d37cefba58f5acaab8d4be7
